packages/gatsby/src/utils/node-manifest.ts

```diff
@@ -0,0 +1,302 @@
+import { ErrorId } from "./../../../gatsby-cli/src/structured-errors/error-map"
+import { getNode } from "./../redux/nodes"
+import { IGatsbyPage, INodeManifest } from "./../redux/types"
+import reporter from "gatsby-cli/lib/reporter"
+import { store } from "../redux/"
+import { internalActions } from "../redux/actions"
+import path from "path"
+import fs from "fs-extra"
+
+interface INodeManifestPage {
+  path?: string
+}
+
+/**
+ * This it the output after processing calls to the public unstable_createNodeManifest action
+ */
+interface INodeManifestOut {
+  page: INodeManifestPage
+  node: {
+    id: string
+  }
+  foundPageBy: FoundPageBy
+}
+
+type FoundPageBy =
+  | `ownerNodeId`
+  | `context.id`
+  | `queryTracking`
+  | `filesystem-route-api`
+  | `none`
+
+/**
+ * Finds a final built page by nodeId
+ *
+ * Note that this function wont work properly in `gatsby develop`
+ * since develop no longer runs all page queries when creating pages.
+ * We use the node id to query mapping to find the right page but
+ * this mapping only exists once you've visited a page in your browser.
+ * When this fn is being used for routing to previews the user wont necessarily have
+ * visited the page in the browser yet.
+ */
+async function findPageOwnedByNodeId({
+  nodeId,
+}: {
+  nodeId: string
+}): Promise<{
+  page: INodeManifestPage
+  foundPageBy: FoundPageBy
+}> {
+  const state = store.getState()
+  const { pages, nodes } = state
+  const { byNode } = state.queries
+
+  // in development queries are run on demand so we wont have an accurate nodeId->pages map until those pages are visited in the browser. We want this mapping before the page is visited in the browser so we can route to the right page in the browser.
+  // So in development we can just use the Map of all pages (pagePath -> pageNode)
+  // but for builds (preview inc builds or regular builds) we will have a full map
+  // of all nodeId's to pages they're queried on and we can use that instead since it
+  // will be a much smaller list of pages, resulting in better performance for large sites
+  const usingPagesMap: boolean =
+    !!process.env._GATSBY_INTERNAL_TEST_NODE_MANIFEST_AS_DEVELOP ||
+    `development` === process.env.NODE_ENV
```

Comment:
You should be able to set process.env.NODE_ENV to 'development' before the test runs and set it back after you're done. Try not to introduce any test code in your production code